iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,460 @@
+                s_defaultOperationTimeout,
+                transportSettings.Proxy,
+                transportSettings.ConnectionLeaseTimeoutMilliseconds);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration)
+        {
+            return CreateAsync(configuration, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Adding configuration: {configuration?.Id}", nameof(CreateAsync));
+
+            try
+            {
+                EnsureInstanceNotClosed();
+
+                if (!string.IsNullOrEmpty(configuration.ETag))
+                {
+                    throw new ArgumentException(ApiResources.ETagSetWhileCreatingConfiguration);
+                }
+
+                var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
+                {
+                    {
+                        HttpStatusCode.PreconditionFailed,
+                        async responseMessage => new PreconditionFailedException(
+                            await ExceptionHandlingHelper.GetExceptionMessageAsync(responseMessage).ConfigureAwait(false))
+                    }
+                };
+
+                return _httpClientHelper.PutAsync(GetConfigurationRequestUri(configuration.Id), configuration, PutOperationType.CreateEntity, errorMappingOverrides, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(CreateAsync)} threw an exception: {ex}", nameof(CreateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Creating configuration: {configuration?.Id}", nameof(CreateAsync));
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the specified Configuration object.
```

Comment:
No need to capitalize "Configuration" here

Reply:
I see this capitalization a lot in this class, so I won't leave a comment on each case